packages/framework/src/Model/Product/Pricing/ProductPriceCalculation.php

```diff
@@ -79,18 +83,19 @@ protected function calculateMainVariantPrice(Product $mainVariant, $domainId, Pr
             $variantPrices[] = $this->calculatePrice($variant, $domainId, $pricingGroup);
         }
 
+        /** @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice $minVariantPrice */
         $minVariantPrice = $this->getMinimumPriceByPriceWithoutVat($variantPrices);
         $from = $this->arePricesDifferent($variantPrices);
 
-        return new ProductPrice($minVariantPrice, $from);
+        return new ProductPrice($minVariantPrice->getPrice(), $from);
```

Comment:
This makes me very unsure about benefit of these changes. We have made those classes final, so they cannot be changed but in the same time we introduced interface to enable the change. I think this two things are in contradiction to each other.

If you do not agree, then at least ProductPrice should accept PriceInterface instead of Price and then you do not need this.

Reply:
all final classes now have their interfaces and the interfaces are used in the code instead of them